tests/tasks/assert_certificate_parameters.yml

```diff
@@ -51,3 +51,20 @@
       - (cert['subject_alt_name'] | sort) == (result.stdout_lines | sort)
     fail_msg: >-
       {{ cert['subject_alt_name'] | sort }} != {{ result.stdout_lines | sort }}
+
+- name: Retrieve key size
+  shell: >-
+    openssl rsa -in {{ cert['key_path'] }} -text -noout |
+    grep -i "private-key" |
+    sed 's/^[^0-9]\+\([0-9]\+\).\+$/\1/'
+  register: result
+  changed_when: false
+  when: cert.get("key_size")
+
+- name: Verify key file attributes
+  assert:
+    that:
+      - (result.stdout | int) == (cert["key_size"] | int)
+    fail_msg: |
+      key_size {{ result.stdout }} != {{ cert['key_size'] | default(2048) }}
+  when: cert.get("key_size")
```

Comment:
Is it necessary to check that the key_size is 2048, even if `cert.key_size` was not specified?

Reply:
so you don't need the `when: cert.get("key_size")` here? or is `cert.key_size` always set?

Reply:
The `when` is there by mistake.